exchanges/oex/oex_types.go

```diff
@@ -0,0 +1,286 @@
+package oex
+
+// ErrCapture helps with error info
+type ErrCapture struct {
+	Error string `json:"code"`
+	Msg   string `json:"msg"`
+}
+
+// TickerDataResponse returns the time and tickerinfo for a trading pair
+type TickerDataResponse struct {
+	High   float64 `json:"high,string"`
+	Volume float64 `json:"vol,string"`
+	Last   float64 `json:"last"`
+	Low    float64 `json:"low,string"`
+	Buy    float64 `json:"buy"`
+	Sell   float64 `json:"sell"`
+	Rose   string  `json:"rose"`
+	Time   int64   `json:"time"`
+}
+
+// TickerResponse returns the time and tickerinfo for a trading pair
+type TickerResponse struct {
+	ErrCapture `json:",omitempty"`
+	Data       TickerDataResponse `json:"data"`
+}
+
+// AllTicker stores the ticker price data for a currency pair
+type AllTicker struct {
+	Symbol string  `json:"symbol"`
+	High   float64 `json:"high,string"`
+	Volume float64 `json:"vol,string"`
+	Last   float64 `json:"last"`
+	Low    float64 `json:"low,string"`
+	Buy    float64 `json:"buy"`
+	Sell   float64 `json:"sell"`
+	Rose   string  `json:"rose"`
+}
+
+// AllTickerDataResponse used for data storage
+type AllTickerDataResponse struct {
+	Date   int64       `json:"date"`
+	Ticker []AllTicker `json:"ticker"`
+}
+
+// AllTickerResponse returns the time and Ticker info for all trading pairs
+type AllTickerResponse struct {
+	ErrCapture `json:",omitempty"`
+	Data       AllTickerDataResponse `json:"data"`
+}
+
+// KlineResponse stores kline info for given currency exchange
+type KlineResponse struct {
+	ErrCapture `json:",omitempty"`
+	Data       [][]float64 `json:"data"`
+}
+
+// TradeData returns market transaction records for a currency pair
+type TradeData struct {
+	Amount    float64 `json:"amount"`
+	Price     float64 `json:"price,string"`
+	ID        int64   `json:"id"`
+	OrderType string  `json:"type"`
+}
+
+// TradeResponse returns market transaction records for a currency pair
+type TradeResponse struct {
+	ErrCapture `json:",omitempty"`
+	Data       []TradeData `json:"data"`
+}
+
+// PairAccuracyResp stores accuracy for a currency pair
+type PairAccuracyResp struct {
+	Symbol          string `json:"symbol"`
+	CountCoin       string `json:"count_coin"`
+	AmountPrecision int64  `json:"amount_precision"`
+	BaseCoin        string `json:"base_coin"`
+	PricePrecision  int64  `json:"price_precision"`
+}
+
+// AllPairResponse stores accuracy for all currency pairs enabled on the exchange
+type AllPairResponse struct {
+	ErrCapture `json:",omitempty"`
+	Data       []PairAccuracyResp `json:"data"`
+}
+
+// CoinInfo stores info about particular coins that user has
+type CoinInfo struct {
+	Coin        string  `json:"coin"`
+	Normal      float64 `json:"normal,string"`
+	Locked      float64 `json:"locked,string"`
+	BtcValuatin string  `json:"btcValuatin"`
```

Comment:
This looks like its current value in BTC we might want this as a float64 as well instead of string